spec/javascripts/angular/services/url_generator_spec.js

```diff
@@ -0,0 +1,27 @@
+//= require spec_helper
+describe('URLGenerator', function() {
+  var urlGenerator;
+  beforeEach(inject(function(_URLGenerator_) {
+    urlGenerator = _URLGenerator_;
+  }));
+
+  it('allows urls with custom paths, no trailing slash', function() {
+    ['http://promdash.server.com/prometheus', 'http://promdash.com'].forEach(function(s) {
+      ['/api/query_range', '/api/query', '/api/metrics', '/arbitrary/endpoint'].forEach(function(ep) {
+        var s = 'http://promdash.server.com/prometheus';
```

Comment:
ahhh this is true. when I expanded the test to have the two different server names, i forgot to delete the original `s` var.